docs/content/guides/migrations/from-step-launchers-to-pipes.mdx

```diff
@@ -0,0 +1,333 @@
+---
+title: "Migrating From Step Launchers to Dagster Pipes | Dagster Docs"
+description: "Learn how to migrate from step launchers to Dagster Pipes in Dagster."
+---
+
+# Migrating From Step Launchers to Dagster Pipes
+
+In this guide, we’ll show you how to migrate from using step launchers to using [Dagster Pipes](/concepts/dagster-pipes) in Dagster.
+
+The reason for this migration and a detailed comparison between the two alternatives can be found in [this GitHub Discussion](https://github.com/dagster-io/dagster/...).
+
+---
+
+## Considerations
+
+When deciding to migrate from step launchers to Dagster Pipes, consider the following:
+
+- **Step launchers** are superceded by Dagster Pipes. While they are still available (and there are no plans for their removal), they are no longer the recommended method for launching external code from Dagster ops and assets. They won't be receiving new features or be under active development.
+- **Dagster Pipes** is a more lightweight and flexible framework, but it does come with a few drawbacks:
+  1. Spark runtime and the code executed will no longer be managed by Dagster for you.
+  2. Dagster Pipes are not compatible with Resources and IO Managers. If you are heavily relying on these features, you might want to keep using step launchers.
+
+## Steps
+
+To migrate from step launchers to Dagster Pipes, you will have to perform the following steps.
+
+### **1. Implement new CI/CD pipelines to prepare your Spark runtime environment**
```

Comment:
Is this step necessary? Personally, I found it a bit overwhelming.

Alternatively, could we just state that the user must install `dagster-pipes` in their execution environment?

Reply:
Previously the Step Launcher was doing all this setup including package preparation for the user. 

I know from personal experience how hard it is to get AWS EMR working so IMO these concrete steps are nice to have, but we can also just link existing EMR Pipes docs.